playwright/_impl/_fetch.py

```diff
@@ -0,0 +1,405 @@
+        )
+
+    async def post(
+        self,
+        url: str,
+        params: Dict[str, Union[bool, float, str]] = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
+        multipart: Dict[str, Union[bytes, bool, float, str, FilePayload]] = None,
+        timeout: float = None,
+        failOnStatusCode: bool = None,
+        ignoreHTTPSErrors: bool = None,
+    ) -> "APIResponse":
+        return await self.fetch(
+            url,
+            method="POST",
+            params=params,
+            headers=headers,
+            data=data,
+            form=form,
+            multipart=multipart,
+            timeout=timeout,
+            failOnStatusCode=failOnStatusCode,
+            ignoreHTTPSErrors=ignoreHTTPSErrors,
+        )
+
+    async def fetch(
+        self,
+        urlOrRequest: Union[str, network.Request],
+        params: Dict[str, Union[bool, float, str]] = None,
+        method: str = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
+        multipart: Dict[str, Union[bytes, bool, float, str, FilePayload]] = None,
+        timeout: float = None,
+        failOnStatusCode: bool = None,
+        ignoreHTTPSErrors: bool = None,
+    ) -> "APIResponse":
+        request = urlOrRequest if isinstance(urlOrRequest, network.Request) else None
+        assert request or isinstance(
+            urlOrRequest, str
+        ), "First argument must be either URL string or Request"
+        assert (
+            len(
+                list(
+                    filter(
+                        lambda entry: entry is True,
+                        [data is not None, form is not None, multipart is not None],
+                    )
+                )
+            )
+            <= 1
+        ), "Only one of 'data', 'form' or 'multipart' can be specified"
+        url = request.url if request else urlOrRequest
+        method = method or (request.method if request else "GET")
+        # Cannot call allHeaders() here as the request may be paused inside route handler.
+        headers_obj = headers or (request.headers if request else None)
+        serialized_headers = serialize_headers(headers_obj) if headers_obj else None
+        json_data = None
+        form_data: Optional[List[NameValue]] = None
+        multipart_data: Optional[List[FormField]] = None
+        post_data_buffer: Optional[bytes] = None
+        if data:
+            if isinstance(data, str):
+                post_data_buffer = data.encode()
+            elif isinstance(data, bytes):
+                post_data_buffer = data
+            elif isinstance(data, (dict, list)):
+                json_data = data
+            else:
+                raise Error(f"Unsupported 'data' type: {type(data)}")
+        elif form:
+            form_data = object_to_array(form)
```

Comment:
object_to_string_array

Reply:
headers have more edge-cases for e.g. set-cookies etc.